app/src/main/java/de/westnordost/streetcomplete/overlays/surface/PathSurfaceOverlayForm.kt

```diff
@@ -0,0 +1,385 @@
+import de.westnordost.streetcomplete.osm.surface.COMMON_SPECIFIC_PAVED_SURFACES
+import de.westnordost.streetcomplete.osm.surface.COMMON_SPECIFIC_UNPAVED_SURFACES
+import de.westnordost.streetcomplete.osm.surface.ParsedCyclewayFootwaySurfacesWithNote
+import de.westnordost.streetcomplete.osm.surface.GENERIC_AREA_SURFACES
+import de.westnordost.streetcomplete.osm.surface.GROUND_SURFACES
+import de.westnordost.streetcomplete.osm.surface.Surface
+import de.westnordost.streetcomplete.osm.surface.SurfaceAndNote
+import de.westnordost.streetcomplete.osm.surface.applyTo
+import de.westnordost.streetcomplete.osm.surface.asItem
+import de.westnordost.streetcomplete.osm.surface.asItemWithFakeNullPossibility
+import de.westnordost.streetcomplete.osm.surface.commonSurfaceObject
+import de.westnordost.streetcomplete.osm.surface.createSurfaceStatus
+import de.westnordost.streetcomplete.osm.surface.shouldBeDescribed
+import de.westnordost.streetcomplete.osm.surface.toItems
+import de.westnordost.streetcomplete.osm.surface.toItemsWithFakeNullPossibility
+import de.westnordost.streetcomplete.overlays.AbstractOverlayForm
+import de.westnordost.streetcomplete.overlays.AnswerItem
+import de.westnordost.streetcomplete.overlays.IAnswerItem
+import de.westnordost.streetcomplete.quests.surface.DescribeGenericSurfaceDialog
+import de.westnordost.streetcomplete.util.getFeatureName
+import de.westnordost.streetcomplete.util.ktx.nonBlankTextOrNull
+import de.westnordost.streetcomplete.view.image_select.DisplayItem
+import de.westnordost.streetcomplete.view.image_select.ImageListPickerDialog
+import de.westnordost.streetcomplete.view.image_select.ItemViewHolder
+
+class PathSurfaceOverlayForm : AbstractOverlayForm() {
+    override val contentLayoutResId = R.layout.fragment_overlay_path_surface_select
+    private val binding by contentViewBinding(FragmentOverlayPathSurfaceSelectBinding::bind)
+
+    private val itemsPerRow = 2
+    /** items to display. May not be accessed before onCreate */
+    val items: List<DisplayItem<Surface?>> = (COMMON_SPECIFIC_PAVED_SURFACES + COMMON_SPECIFIC_UNPAVED_SURFACES + GROUND_SURFACES + GENERIC_AREA_SURFACES).toItemsWithFakeNullPossibility()
+    private val cellLayoutId: Int = R.layout.cell_labeled_icon_select
+    private var originalSurfaceStatus: ParsedCyclewayFootwaySurfacesWithNote? = null
+    private var isSegregatedLayout = false
+
+    private var selectedStatusForMainSurface: DisplayItem<Surface?>? = null
+        set(value) {
+            field = value
+            updateSelectedCell()
+        }
+    private var selectedStatusForCyclewaySurface: DisplayItem<Surface?>? = null
+        set(value) {
+            field = value
+            updateSelectedCell()
+        }
+    private var selectedStatusForFootwaySurface: DisplayItem<Surface?>? = null
+        set(value) {
+            field = value
+            updateSelectedCell()
+        }
+
+        override val otherAnswers: List<IAnswerItem> get() {
```

Comment:
Starting from this line, the indentation in the file is messed up.

Also, this function can be body-less.

Reply:
curiously ktlint is not complaining

Reply:
We disabled the `indent` rule:

https://github.com/streetcomplete/StreetComplete/blob/270cedfcd7eb370655454dc4ce16ed880dd47827/.editorconfig#L19